### DISCUSSION TITLE
Custom dependency graph format for unpublished / GitHub-only repositories

### DISCUSSION THREAD

**Discussion by TheLartians** [community/community#23390] [New to GitHub]
<p>Hey everyone!</p>
<p>I’ve been working on a <a href="https://github.com/cpm-cmake/CPM.cmake" rel="noopener nofollow ugc">very simple dependency manager for C++</a> that is able to fetch dependencies on-demand from Git repositories (and caches them to save bandwidth). We are currently exploring the possibility to auto-generate manifest files for tracking dependencies in the GitHub dependency graph. So far we’ve tried listing the dependencies in an <a href="https://github.com/alandefreitas/requirements_test/blob/62b0af1a8026d1e74c8f346b76e5cb063cddd023/package-lock.json" rel="noopener nofollow ugc">npm-style <code>package.json</code> file</a>, which seems to be recognised, however the according dependencies appear grayed out, probably as the packages aren’t published on npm.</p>
<p><img src="https://user-images.githubusercontent.com/4437447/181076713-c97f6e38-94fe-4125-80e5-db54e90276c9.png" alt="image" data-base62-sha1="5khdUO1uYomDxnrMmBZHr1LN0jM" width="618" height="210"></p>
<p>Is there a compatible format that we can use to link to generic GitHub repositories that can be properly parsed and linked by the dependency graph?</p>
<p>Thanks,</p>
<p>Lars</p>

**Comment by ernest-phillips** [comment, marked as answer]
<p>Hi <a class="mention" href="/thelartians">@TheLartians</a> thanks for making your first post in the Community forum!</p>
<p>Your assumptions are correct here. If the package isn’t published to NPM, dependabot will not be able to make the associated connection to the proper repository. We do not have a way to get those updates to it when changes are made.</p>
<p>There’s no workaround for this one either, I’m afraid.</p>

**Comment by TheLartians** [comment]
<p>Got it, thanks for clarifying!</p>
